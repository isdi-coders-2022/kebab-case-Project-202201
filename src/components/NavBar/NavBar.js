import styled from "styled-components";

const NavBarStyled = styled.div`
  position: fixed;
  bottom: 0;
  width: 100%;
  height: 65px;
  background: #461094;
  box-shadow: 0px -4px 4px rgba(0, 0, 0, 0.25);
  display: flex;
  justify-content: space-around;
`;

const NavBar = () => {
  return (
    <>
      <NavBarStyled>
        <button>hola</button>
        <button>hola</button>
        <button>hola</button>
      </NavBarStyled>
    </>
  );
};

export default NavBar;
