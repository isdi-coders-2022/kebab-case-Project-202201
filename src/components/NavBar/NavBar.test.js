import { render, screen } from "@testing-library/react";
import TestRenderer from "react-test-renderer";
import NavBar from "./NavBar";

describe("Given a NavBar component", () => {
  describe("When it get a const newNavBar that creates NavBar", () => {
    test("then it should create and compare a snapshot with newNavBar", () => {
      const newNavBar = TestRenderer.create(<NavBar />);
      expect(newNavBar.toJSON()).toMatchSnapshot();
    });
  });

  describe("WWhen it is rendered", () => {
    test("Then it should render 3 buttons on screen", () => {
      const expectedNoOfButtons = 3;
      render(<NavBar />);
      const buttons = screen.getAllByRole("button");

      expect(buttons.length).toBe(expectedNoOfButtons);
    });
  });
});
