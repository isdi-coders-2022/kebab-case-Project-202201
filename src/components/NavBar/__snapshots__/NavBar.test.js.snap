// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a NavBar component When it get a const newNavBar that creates NavBar then it should create and compare a snapshot with newNavBar 1`] = `
<div
  className="sc-jRQBWg fiajYn"
>
  <button
    className="sc-bdvvtL jbnYpL"
    onClick={[Function]}
    type="button"
  >
    HOME
  </button>
  <button
    className="sc-gsDKAQ dQpKYz"
    onClick={[Function]}
    type="button"
  >
    <img
      alt="multidurum"
      src="img/multidurum.png"
    />
  </button>
  <button
    className="sc-bdvvtL jbnYpL"
    onClick={[Function]}
    type="button"
  >
    FAVS
  </button>
</div>
`;
